[2022-06-10 17:21:09,764] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: spark_pipeline_dag.extract_data_task manual__2022-06-10T14:20:43.516527+00:00 [queued]>
[2022-06-10 17:21:09,766] {taskinstance.py:1159} INFO - Dependencies all met for <TaskInstance: spark_pipeline_dag.extract_data_task manual__2022-06-10T14:20:43.516527+00:00 [queued]>
[2022-06-10 17:21:09,766] {taskinstance.py:1356} INFO - 
--------------------------------------------------------------------------------
[2022-06-10 17:21:09,766] {taskinstance.py:1357} INFO - Starting attempt 1 of 2
[2022-06-10 17:21:09,766] {taskinstance.py:1358} INFO - 
--------------------------------------------------------------------------------
[2022-06-10 17:21:09,770] {taskinstance.py:1377} INFO - Executing <Task(PythonOperator): extract_data_task> on 2022-06-10 14:20:43.516527+00:00
[2022-06-10 17:21:09,771] {standard_task_runner.py:52} INFO - Started process 1063 to run task
[2022-06-10 17:21:09,773] {standard_task_runner.py:79} INFO - Running: ['airflow', 'tasks', 'run', 'spark_pipeline_dag', 'extract_data_task', 'manual__2022-06-10T14:20:43.516527+00:00', '--job-id', '32', '--raw', '--subdir', 'DAGS_FOLDER/spark_pipeline_dag.py', '--cfg-path', '/var/folders/7p/cr8xz9ln34l55g9c5fr8glxm0000gn/T/tmpz_97yxkx', '--error-file', '/var/folders/7p/cr8xz9ln34l55g9c5fr8glxm0000gn/T/tmprx7g30q9']
[2022-06-10 17:21:09,774] {standard_task_runner.py:80} INFO - Job 32: Subtask extract_data_task
[2022-06-10 17:21:09,789] {task_command.py:370} INFO - Running <TaskInstance: spark_pipeline_dag.extract_data_task manual__2022-06-10T14:20:43.516527+00:00 [running]> on host macbook-air-german.local
[2022-06-10 17:21:09,809] {taskinstance.py:1569} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=spark_pipeline_dag
AIRFLOW_CTX_TASK_ID=extract_data_task
AIRFLOW_CTX_EXECUTION_DATE=2022-06-10T14:20:43.516527+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-06-10T14:20:43.516527+00:00
[2022-06-10 17:21:22,587] {python.py:173} INFO - Done. Returned value was: DataFrame[datetime: string, milliseconds: string, site_id: string, sitezone_id: string, user_id: string, google_id_first: string, yandex_id_first: string]
[2022-06-10 17:21:22,621] {xcom.py:584} ERROR - Could not serialize the XCom value into JSON. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your airflow config.
[2022-06-10 17:21:22,622] {taskinstance.py:1889} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/germangerken/PycharmProjects/CyberBrainExs/venv/lib/python3.10/site-packages/airflow/utils/session.py", line 68, in wrapper
    return func(*args, **kwargs)
  File "/Users/germangerken/PycharmProjects/CyberBrainExs/venv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 2385, in xcom_push
    XCom.set(
  File "/Users/germangerken/PycharmProjects/CyberBrainExs/venv/lib/python3.10/site-packages/airflow/utils/session.py", line 68, in wrapper
    return func(*args, **kwargs)
  File "/Users/germangerken/PycharmProjects/CyberBrainExs/venv/lib/python3.10/site-packages/airflow/models/xcom.py", line 191, in set
    value = cls.serialize_value(
  File "/Users/germangerken/PycharmProjects/CyberBrainExs/venv/lib/python3.10/site-packages/airflow/models/xcom.py", line 582, in serialize_value
    return json.dumps(value).encode('UTF-8')
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type DataFrame is not JSON serializable
[2022-06-10 17:21:22,624] {taskinstance.py:1395} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_pipeline_dag, task_id=extract_data_task, execution_date=20220610T142043, start_date=20220610T142109, end_date=20220610T142122
[2022-06-10 17:21:22,627] {standard_task_runner.py:92} ERROR - Failed to execute job 32 for task extract_data_task (Object of type DataFrame is not JSON serializable; 1063)
[2022-06-10 17:21:22,669] {local_task_job.py:156} INFO - Task exited with return code 1
[2022-06-10 17:21:22,676] {dagrun.py:647} WARNING - Failed to get task '<TaskInstance: spark_pipeline_dag.spark_pipeline_task manual__2022-06-10T14:20:43.516527+00:00 [removed]>' for dag 'spark_pipeline_dag'. Marking it as removed.
[2022-06-10 17:21:22,679] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
